lib/node_modules/@stdlib/_tools/remark/plugins/remark-lint-expected-html-sections/test/test.js

```diff
@@ -0,0 +1,165 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var join = require( 'path' ).join;
+var tape = require( 'tape' );
+var remark = require( 'remark' );
+var readFileSync = require( '@stdlib/fs/read-file' ).sync;
+var expectedSections = require( './../lib' );
+
+
+// VARIABLES //
+
+var TEST_FIXTURES = join( __dirname, 'fixtures' );
+var CUSTOM_SCHEMA = {
+	'root': {
+		'required': [ 'usage', 'examples', 'links', 'related' ],
+		'optional': [ 'intro', 'notes', 'c', 'references' ]
+	}
+};
+var FIXTURE_VALID_COMPLETE = readFileSync( join( TEST_FIXTURES, 'valid_complete.md.txt' ), 'utf8' ).toString();
+var FIXTURE_VALID_MEMOIZE = readFileSync( join( TEST_FIXTURES, 'valid_memoize.md.txt' ), 'utf8' ).toString();
+var FIXTURE_VALID_ASIN = readFileSync( join( TEST_FIXTURES, 'valid_asin.md.txt' ), 'utf8' ).toString();
+var FIXTURE_INVALID_ASIN = readFileSync( join( TEST_FIXTURES, 'invalid_asin.md.txt' ), 'utf8' ).toString();
+var FIXTURE_MISSING_ROOT = readFileSync( join( TEST_FIXTURES, 'missing_required_root.md.txt' ), 'utf8' ).toString();
+var FIXTURE_INCOMPLETE_C = readFileSync( join( TEST_FIXTURES, 'incomplete_c.md.txt' ), 'utf8' ).toString();
+
+
+// FUNCTIONS //
+
+function lint( text, options, clbk ) {
+	remark().use( expectedSections, options ).process( text, clbk );
+}
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof expectedSections, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'validates valid_complete.md.txt with default schema', function test( t ) {
+	lint( FIXTURE_VALID_COMPLETE, {}, done );
+
+	function done( err, file ) {
+		if ( err ) {
+			t.fail( err.message );
+		}
+		t.strictEqual( file.messages.length, 0, 'has no errors for valid fixture with all sections' );
+		t.end();
+	}
+});
+
+tape( 'validates valid_memoize.md.txt', function test( t ) {
+	lint( FIXTURE_VALID_MEMOIZE, {}, done );
+
+	function done( err, file ) {
+		if ( err ) {
+			t.fail( err.message );
+		}
+		t.strictEqual( file.messages.length, 0, 'has no errors for memoize README fixture' );
+		t.end();
+	}
+});
+
+tape( 'validates valid_asin.md.txt with C section', function test( t ) {
+	lint( FIXTURE_VALID_ASIN, {}, done );
+
+	function done( err, file ) {
+		if ( err ) {
+			t.fail( err.message );
+		}
+		t.strictEqual( file.messages.length, 0, 'has no errors for asin README fixture with C section' );
+		t.end();
+	}
+});
+
+tape( 'detects missing C usage section in invalid_asin.md.txt', function test( t ) {
+	lint( FIXTURE_INVALID_ASIN, {}, done );
+
+	function done( err, file ) {
+		if ( err ) {
+			t.fail( err.message );
+		}
+		t.strictEqual( file.messages.length, 1, 'has one error for invalid asin fixture' );
+		t.strictEqual( file.messages[0].reason.includes('Missing required sections in "c" section: `usage`'), true, 'error identifies the missing C usage section' );
```

Comment:
Refactor to avoid use of `includes` here and below. Also, spacing.

Reply:
Addressed.